ast/src/parsed/display.rs

```diff
@@ -892,4 +939,124 @@ mod tests {
         assert_eq!(p.clone().join(s.clone()).to_string(), "::abc::y");
         assert_eq!(SymbolPath::from(p.join(s)).to_string(), "::abc::y");
     }
+
+    mod binary_op_parentheses {
```

Comment:
Can you turn all these tests into the other form please and then remove this? These tests are very verbose and we need to modify all of them if we change the structs.